app/controllers/events_controller.rb

```diff
@@ -76,7 +75,18 @@ def update
     end
   end
 
+  def destroy
+    @event.skip_validation = true
+    delete_event_data
+    @event.applications.delete_all
```

Comment:
I think we should not delete the applications but rather do the same approach as with delete_event_data. for the same reasons as we want to keep the ids, countries and created_at of the events.

Maybe we can add a private method as well like:

def delete_application_data
     attributes = @event.applications.attributes.keys - ["id", "event_id", "created_at", "updated_at", "submitted", 
     "deleted"]
     columns = {deleted: true}
     attributes.each do |attr|
         if @event.applications.[attr].class == TrueClass || @event.applications.[attr].class == FalseClass
           columns[attr] = false
         else
           columns[attr] = nil
        end
      end
     @event.applications.update_attributes(columns)
 end